### PR TITLE
Fix FLIRT v5 ##signatures

### LINKED ISSUE #12420: IDA .sig files with version 5 compression not supported
The comments in this similar issue seem to imply this was fixed: https://github.com/radare/radare2/issues/3665 but I'm still getting the same error:

```
[0x00000000]> zfs **.sig
Sorry we do not support the signatures version 5 compression.
We encountered an error while parsing the file ./pc/bcb5rt.sig. Sorry.
```

The PR that fixed the compression issue was apparently in zip.c, but it doesn't look like any change was made to flirt.c which throws this error whenever a flirt sig with version 5 compression is detected.

### PR DESCRIPTION
**Pull request by swoops**
**Checklist**

- [x] Closing issues: #issue 
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

Adds FLIRT v5 support to r2. Based off of rizin code. I have not tested this, but I figured I would go ahead and share. This closes #12420.

Also, clang format was giving me some strange output. It wanted me to add an extra tab before `r_inflate`. So double check I did not do something stupid, like miss a closing `)` or `}`.

### CODE REVIEW

File: libr/anal/flirt.c [hunk 1]
```diff
@@ -1371,12 +1368,18 @@ static RFlirtNode *flirt_parse(const RAnal *anal, RBuffer *flirt_buf) {
...
 	if (header->features & IDASIG__FEATURE__COMPRESSED) {
-		if (version == 5) {
-			eprintf ("Sorry we do not support the signatures version 5 compression.\n");
-			goto exit;
-		}
-		if (!(decompressed_buf = r_inflate (buf, size, NULL, &decompressed_size))) {
-			eprintf ("Decompressing failed.\n");
+		if (version >= 5 && version < 7) {
+			if (!(decompressed_buf = r_inflate_ignore_header (buf, size, NULL, &decompressed_size))) {
+				eprintf ("Decompressing failed.\n");
```
Comment (trufae) [inline libr/anal/flirt.c:1373]:
```suggestion
				eprintf ("Decompression failed.\n");
```

File: libr/anal/flirt.c [hunk 2]
```diff
@@ -1371,12 +1368,18 @@ static RFlirtNode *flirt_parse(const RAnal *anal, RBuffer *flirt_buf) {
...
+				goto exit;
+			}
+		} else if (version >= 7) {
+			if (!(decompressed_buf = r_inflate (buf, size, NULL, &decompressed_size))) {
+				eprintf ("Decompressing failed.\n");
```
Comment (trufae) [inline libr/anal/flirt.c:1378]:
```suggestion
				eprintf ("Decompression failed.\n");
```

File: libr/anal/flirt.c [hunk 3]
```diff
@@ -1371,12 +1368,18 @@ static RFlirtNode *flirt_parse(const RAnal *anal, RBuffer *flirt_buf) {
...
+				goto exit;
+			}
+		} else {
+			eprintf ("Sorry we do not support the signatures version %c compression.\n", version);
```
Comment (trufae) [inline libr/anal/flirt.c:1382]:
```suggestion
			eprintf ("FLIRT signatures version %d is not supported.\n", version);
```

File: libr/anal/flirt.c [hunk 4]
```diff
@@ -1371,12 +1368,18 @@ static RFlirtNode *flirt_parse(const RAnal *anal, RBuffer *flirt_buf) {
 	}
 
 	if (header->features & IDASIG__FEATURE__COMPRESSED) {
-		if (version == 5) {
```
Comment (trufae) [inline libr/anal/flirt.c:1374]: check for null in the line 1368 (buf = malloc (size)) above ^

File: libr/anal/flirt.c [hunk 5]
```diff
@@ -1371,12 +1368,18 @@ static RFlirtNode *flirt_parse(const RAnal *anal, RBuffer *flirt_buf) {
...
-			eprintf ("Sorry we do not support the signatures version 5 compression.\n");
-			goto exit;
-		}
-		if (!(decompressed_buf = r_inflate (buf, size, NULL, &decompressed_size))) {
-			eprintf ("Decompressing failed.\n");
+		if (version >= 5 && version < 7) {
+			if (!(decompressed_buf = r_inflate_ignore_header (buf, size, NULL, &decompressed_size))) {
```
Comment (trufae) [inline libr/anal/flirt.c:1372]: i think `dbuf` for varname is quite better for reading

File: libr/util/zip.c [hunk 6]
```diff
@@ -23,7 +23,16 @@ static const char *gzerr(int n) {
...
+/**
+ * \brief inflate zlib compressed or gzipped.
+ * \param src source compressed bytes
+ * \param srcLen source bytes length
+ * \param srcConsumed comsumed source bytes length
+ * \param dstLen uncompressed uncompressed bytes length
+ * \param controls the size of the history buffer (or “window size”), and what header and trailer format is expected.
+ * \return ptr to uncompressed
+*/
+static ut8 *r_inflatew(const ut8 *src, int srcLen, int *srcConsumed, int *dstLen, int wbits) {
```
Comment (trufae) [inline libr/util/zip.c:35]: Some things that can be improved here:

* we dont use doxygen comments, so feel free to remove them
* Dont use camelcase
* buffers length should use size_t. int have sign and size problems

File: libr/util/zip.c [hunk 7]
```diff
@@ -84,3 +92,21 @@ R_API ut8 *r_inflate(const ut8 *src, int srcLen, int *srcConsumed, int *dstLen)
...
+R_API ut8 *r_inflate(const ut8 *src, int srcLen, int *srcConsumed, int *dstLen) {
+	return r_inflatew (src, srcLen, srcConsumed, dstLen, MAX_WBITS + 32);
+}
+
+/**
+ * @brief inflate zlib compressed or gzipped. The input must be a raw stream
+ * with no header or trailer.
+ * @see r_inflatew()
+*/
+R_API ut8 *r_inflate_ignore_header(const ut8 *src, int srcLen, int *srcConsumed, int *dstLen) {
```
Comment (trufae) [inline libr/util/zip.c:110]: what about `r_inflate_raw` ?

**Review by trufae — CHANGES_REQUESTED** [review]
ill try to provide a test binary for this thing. thanks!

File: libr/util/zip.c (hunk 6 above)
Comment (swoops) [reply libr/util/zip.c]: I think I took care of this. Unless I missed a buffer length somewhere.

**Pull request merged**